cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -398,6 +403,93 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	return nil
 }
 
+func (a *StaticAutoscaler) handleQuoteAndStockoutErrors(currentTime time.Time) {
+	cloudProviderNodeInstances := a.clusterStateRegistry.GetCloudProviderNodeInstances()
+
+	nodeGroups := a.CloudProvider.NodeGroups()
+	for _, nodeGroup := range nodeGroups {
+		instances, found := cloudProviderNodeInstances[nodeGroup.Id()]
+		if !found {
+			continue
+		}
+
+		outOfResourcesInstancesByErrorCode := make(map[string][]cloudprovider.Instance)
+		for _, instance := range instances {
+			if instance.Status != nil && instance.Status.State == cloudprovider.StateCreating && instance.Status.ErrorInfo != nil {
+				errorInfo := instance.Status.ErrorInfo
+				if errorInfo.ErrorClass == cloudprovider.ErrorOutOfResources {
+					errorInstances, found := outOfResourcesInstancesByErrorCode[errorInfo.ErrorCode]
```

Comment:
why not just:

`errorInstancesByCode[errorInfo.ErrorCode] = append(errorInstancesByCode[errorInfo.ErrorCode], errorInstance)`